### PR TITLE
MirAL version 2.0

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Tidy up the MirAL API (breaking the ABI in the process)

### CODE REVIEW

**Comment by AlanGriffiths** [comment]
This has broken the tiling WMP. (The fix should be localized, so I'll leave this up for now.)

File: include/miral/miral/version.h [hunk 1]
```diff
@@ -31,7 +31,7 @@
  *
  * See also: http://semver.org/
  */
-#define MIRAL_MAJOR_VERSION (1)
+#define MIRAL_MAJOR_VERSION (3)
```
Comment (RAOF) [inline include/miral/miral/version.h]: I think you mean `#define MIRAL_MAJOR_VERSION (2)` here, right?

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -1,8 +1,8 @@
 # We need MIRPLATFORM_ABI in both libmirplatform and the platform implementations.
 set(MIRPLATFORM_ABI 16)
 
-set(MIRAL_VERSION_MAJOR 1)
-set(MIRAL_VERSION_MINOR 5)
+set(MIRAL_VERSION_MAJOR 3)
```
Comment (RAOF) [inline src/CMakeLists.txt]: Again, `set(MIRAL_VERSION_MAJOR 2)`?

**Comment by Saviq** [comment]
Do we need a `MIRAL_VERSION` file next to `VERSION` as well?

**Comment by AlanGriffiths** [comment]
> @Saviq Do we need a MIRAL_VERSION file next to VERSION as well?

I was assuming we'd put both MIR_VERSION* and MIRAL_VERSION* in VERSION.

**Comment by AlanGriffiths** [comment]
> I've made a QtMir branch to test against this:
https://github.com/gerboland/qtmir/tree/mir-0.30-compat
It works with the demo server, I've not tested unity8 tho. I'm happy enough with the API changes to give it my +1

Shouldn't that be "mir-0.31-compat"?

**Comment by gerboland** [comment]
Yes, branch name not reflecting contents

**Comment by Saviq** [comment]
bors r=RAOF,gerboland,Saviq

**Comment by AlanGriffiths** [comment]
> /usr/lib/aarch64-linux-gnu/glib-2.0/glib-compile-schemas: 2: /usr/lib/aarch64-linux-gnu/glib-2.0/glib-compile-schemas: Syntax error: "&" unexpected
> 
> /usr/lib/aarch64-linux-gnu/glib-2.0/gio-querymodules: 17: /usr/lib/aarch64-linux-gnu/glib-2.0/gio-querymodules: Syntax error: EOF in backquote substitution

Why would that be happening on 16.04?

**Comment by AlanGriffiths** [comment]
bors r=RAOF,gerboland,Saviq

**Pull request merged**